internal/api/approval/state.go

```diff
@@ -87,41 +77,85 @@ func (s *state) addIgnoredReviewers(ignoredTeamMembers []string) {
 	s.ignoredReviewers = uniqueAppend(s.ignoredReviewers, ignoredReviewers)
 }
 
+func (s *state) allRulesFulfilled() bool {
+	for _, rule := range s.matchedRules {
+		if !rule.Fulfilled() {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (s *state) pendingTeamNames() []string {
+	allPending := []string{}
```

Comment:
isn't it more go-idiomatic to not instantiate until necessary :)?
```suggestion
	var allPending []string = nil
```

Reply:
I'm sure you're right but I need empty list to return as default in case there are no pending team names.